### PR TITLE
Extract temp

### PR DESCRIPTION
**Pull request by SandraSamardzic**
Migrate extract temp to new architecture

### CODE REVIEW

File: src/Refactoring-Core/RBExtractToTemporaryRefactoring.class.st [hunk 1]
```diff
@@ -42,14 +43,13 @@ RBExtractToTemporaryRefactoring class >> model: aRBSmalltalk extract: anInterval
...
-		  (RBCondition withBlock: [
-			   self verifySelectedInterval.
-			   self checkVariableName.
-			   true ]) }
+		  (ReDefinesSelectorsCondition new definesSelectors: (Array with: selector) in: class).
+		  (ReIsValidInstanceVariableName new name: newVariableName).
+		  (ReBlockCondition new 
+				block: [
+					self verifySelectedInterval.
+			   		self checkVariableName.
```
Comment (balsa-sarenac) [inline src/Refactoring-Core/RBExtractToTemporaryRefactoring.class.st]: Would it make sense to create condition classes for these two conditions as well?
- `verifySelectedInterval` just checks if the selection is a sequence,
- `checkVariableName` should be already defined as a condition.
I suggest at least to inline `verifySelectedInterval` and add `ReCheckVariableName` to the list of preconditions.

**Review by balsa-sarenac — CHANGES_REQUESTED** [review]
Nice work, just a minor comment. If you decide on creating a condition for checking if selection is a sequence, please do include tests for it, if not, we will deal with it in a latter PR.

**Comment by Ducasse** [comment]
Thanks sandra 
This is really great to see all these enhancements.

**Comment by Ducasse** [comment]
I see that the following test is not passing 

[osx-64 / Tests-osx-64 / MacOSX64.Refactoring.Core.Tests.ReMethodsHaveNoDuplicatesTest.testMethodDuplicatedInSibling](https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-17153/2/testReport/junit/MacOSX64.Refactoring.Core.Tests/ReMethodsHaveNoDuplicatesTest/osx_64___Tests_osx_64___testMethodDuplicatedInSibling/)

**Comment by SandraSamardzic** [comment]
Thank you @Ducasse, I am trying my best. I did see that the test didn't pass but I don't see where is the test nor where is the problem...

**Comment by Ducasse** [comment]
I will integrate and we can see. 
You cannot find `ReMethodsHaveNoDuplicatesTest >> testMethodDuplicatedInSibling`

**Comment by balsa-sarenac** [comment]
@Ducasse I think she couldn't find failing tests in the CI. I'll send her a message with some instructions

**Comment by balsa-sarenac** [comment]
@Ducasse I think that failing test is from Caro's PR, this can be merged

**Review by balsa-sarenac — APPROVED** [review]

**Pull request merged**